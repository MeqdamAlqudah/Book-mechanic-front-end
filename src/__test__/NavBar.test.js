import * as React from 'react';
import { Link, BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../components/Pages/NavBar';

import store from '../redux/configureStore';

const renderer = require('react-test-renderer');

global.React = React;
describe('Test NavBar', () => {
  test('test Link path', () => {
    const clickHandler = () => {};
    const component = renderer.create(
      <Router location="/">
        <Link to="/" clickHandler={clickHandler}>go to</Link>
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree.props.href).toEqual('/');
  });
  test('test Navbar component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router location="/">
          <Navbar />
        </Router>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree.props.href).toMatchSnapshot();
  });
});
