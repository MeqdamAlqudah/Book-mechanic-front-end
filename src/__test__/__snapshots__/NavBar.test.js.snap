// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NavBar test Navbar component 1`] = `undefined`;
